### PR TITLE
Clipboard Rotation

### PR DESCRIPTION
**Pull request by CodedStingray**

### CODE REVIEW

File: src/main/java/net/codedstingray/worldshaper/util/math/MathUtils.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package net.codedstingray.worldshaper.util.math;
+
+public class MathUtils {
+
+    public static int posMod(int number, int divider) {
```
Comment (ZidArgs) [inline src/main/java/net/codedstingray/worldshaper/util/math/MathUtils.java:5]: could use documentation to describe what this does. its obvious from the code but not when you may use this

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java [hunk 2]
```diff
@@ -39,4 +40,82 @@ public static Vector3i getDirectionVectorByName(String name) {
...
+
+    public static Vector3f createVector3f(Vector3i v) {
+        return new Vector3fi(v.getX(), v.getY(), v.getZ());
+    }
+
+    public static Vector3f createBlockVector3f(Vector3i v) {
+        return new Vector3fi(v.getX() + 0.5f, v.getY() + 0.5f, v.getZ() + 0.5f);
+    }
+
+    public static Vector3i getBoundingBoxMinVector(Vector3i... vectors) {
```
Comment (ZidArgs) [inline src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:74]: could use documentation

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java [hunk 3]
```diff
@@ -39,4 +40,82 @@ public static Vector3i getDirectionVectorByName(String name) {
...
+        Arrays.stream(vectors).forEach(v -> {
+            min.x = Math.min(min.x, v.getX());
+            min.y = Math.min(min.y, v.getY());
+            min.z = Math.min(min.z, v.getZ());
+        });
+
+        return min.toImmutable();
+    }
+
+    public static Vector3i getBoundingBoxMaxVector(Vector3i... vectors) {
```
Comment (ZidArgs) [inline src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:86]: could use documentation

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java [hunk 4]
```diff
@@ -39,4 +40,82 @@ public static Vector3i getDirectionVectorByName(String name) {
...
+        Arrays.stream(vectors).forEach(v -> {
+            max.x = Math.max(max.x, v.getX());
+            max.y = Math.max(max.y, v.getY());
+            max.z = Math.max(max.z, v.getZ());
+        });
+
+        return max.toImmutable();
+    }
+
+    public static Vector3f getBoundingBoxMinVector(Vector3f... vectors) {
```
Comment (ZidArgs) [inline src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:98]: could use documentation

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java [hunk 5]
```diff
@@ -39,4 +40,82 @@ public static Vector3i getDirectionVectorByName(String name) {
...
+        Arrays.stream(vectors).forEach(v -> {
+            min.x = Math.min(min.x, v.getX());
+            min.y = Math.min(min.y, v.getY());
+            min.z = Math.min(min.z, v.getZ());
+        });
+
+        return min.toImmutable();
+    }
+
+    public static Vector3f getBoundingBoxMaxVector(Vector3f... vectors) {
```
Comment (ZidArgs) [inline src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:110]: could use documentation

**Review by ZidArgs — CHANGES_REQUESTED** [review]

File: src/main/java/net/codedstingray/worldshaper/util/math/MathUtils.java (hunk 1 above)
Comment (CodedStingray) [reply src/main/java/net/codedstingray/worldshaper/util/math/MathUtils.java:5]: As discussed, I will add proper documentation to the entire plugin code in a dedicated PR

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java (hunk 2 above)
Comment (CodedStingray) [reply src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:74]: As discussed, I will add proper documentation to the entire plugin code in a dedicated PR

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java (hunk 3 above)
Comment (CodedStingray) [reply src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:86]: As discussed, I will add proper documentation to the entire plugin code in a dedicated PR

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java (hunk 4 above)
Comment (CodedStingray) [reply src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:98]: As discussed, I will add proper documentation to the entire plugin code in a dedicated PR

File: src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java (hunk 5 above)
Comment (CodedStingray) [reply src/main/java/net/codedstingray/worldshaper/util/vector/VectorUtils.java:110]: As discussed, I will add proper documentation to the entire plugin code in a dedicated PR

**Review by ZidArgs — APPROVED** [review]

**Pull request merged**